### PR TITLE
ui: Auth Methods -  Namespace rules tab

### PR DESCRIPTION
**Pull request by kaxcode**
- Added mock data for NamespaceRules
- Created NamespaceTable with styling
- Created a Namespace rules tab with a flag to only show the tab if namespaces are enabled

[Demo Link
](https://consul-ui-staging-191qln2jp-hashicorp.vercel.app/ui/dc1/acls/auth-methods/alarm-6/namespace-rules#CONSUL_ACLS_ENABLE=1)

Screen Shot 2021-03-26 at 4 00 01 PM

### CODE REVIEW

File: ui/packages/consul-ui/app/components/consul/auth-method/index.scss [hunk 1]
```diff
@@ -24,9 +24,17 @@
         font-weight: $typo-weight-semibold;
         font-size: $typo-size-700;
       }
-      tbody td {
-        font-size: $typo-size-600;
-        color: $black;
+      tbody {
```
Comment (kaxcode) [inline ui/packages/consul-ui/app/components/consul/auth-method/index.scss:27]: @johncowen I thought about making this into a styling component but this is an overwrite of the already existing %table. Later on I will take a look at what can be refactored in %table to delete all this. For now, I'm just overwriting to get this working.

File: ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show.hbs [hunk 2]
```diff
@@ -27,6 +27,9 @@
       compact
         (array
           (hash label="General info" href=(href-to "dc.acls.auth-methods.show.auth-method") selected=(is-href "dc.acls.auth-methods.show.auth-method"))
+    (if (env "CONSUL_NSPACES_ENABLED")
```
Comment (kaxcode) [inline ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show.hbs:30]: @johncowen Is this the right conditional? The demo site had it showing before I changed the scenario to be nspaces enabled. Does the demo site always have nspaces enabled?

File: ui/packages/consul-ui/app/components/consul/auth-method/index.scss (hunk 1 above)
Comment (johncowen) [reply ui/packages/consul-ui/app/components/consul/auth-method/index.scss:27]: Sounds great!

File: ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show.hbs (hunk 2 above)
Comment (johncowen) [reply ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show.hbs:30]: Yeah this is the right env var to use. You're right though, the demo site does have everything enabled by default. We should maybe change that in future now that we have our bookmarklets and URLs to be able to link to specific scenarios.

I mentioned a while back about using potentially using `can` for this type of feature flagging I saw some convo about others specifically not doing that, so maybe we should chat about it before we 100% decide.

TL;DR The conditional here is 💯

File: ui/packages/consul-ui/app/router.js [hunk 3]
```diff
@@ -199,6 +199,9 @@ export const routes = {
           'binding-rules': {
             _options: { path: '/binding-rules' },
           },
+          'namespace-rules': {
+            _options: { path: '/namespace-rules' },
```
Comment (johncowen) [inline ui/packages/consul-ui/app/router.js:203]: Bit of a quibble this one. The URL here is great, although we refer to any Routes that are namespace related as `nspace`. So maybe we should call this one `nspace-rules`. 

I'm trying to imagine myself writing this in amongst some code and wondering which one I would write automatically without thinking `nspace-rules` or `namespace-rules` - as I've trained myself in this codebase to never write the word `namespace` I'd probably go with `nspace-rules`.

File: ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show/namespace-rules.hbs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+
+<div class="tab-section">
+{{#if (gt item.NamespaceRules.length 0)}}
+  <p>A set of rules that can control which namespace tokens created via this auth method will be created within. Unlike binding rules, the first matching namespace rule wins.</p>
```
Comment (johncowen) [inline ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show/namespace-rules.hbs:4]: I think this `<p>` tag needs to show whether you have rules or not, check with design maybe but in other places in the UI we still show the info paragraphs at the top if there are no results to show.

File: ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show/namespace-rules.hbs [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  <Consul::AuthMethod::NamespaceTable @items={{item.NamespaceRules}} />
```
Comment (johncowen) [inline ui/packages/consul-ui/app/templates/dc/acls/auth-methods/show/namespace-rules.hbs:5]: Here's another thing that is a little difficult to give reasons/rules for naming things. 

Ideally this one would be `Consul::AuthMethod::NamespaceRule::List`. List here doesn't necessarily mean 'this is formatted using `li`s' more that this is multiple of them rather than a single one. Then if we ever need to change how we markup the `NamespaceRule::List` (so not use tables anymore), we don't have to rename all the areas in the code where we use the component.

**Review by johncowen — APPROVED** [review]
🔥 This is sooo good! 👏 

I left a few comments, and I think the only definite change we should make is to make that info paragraph of text show even when you have no rules.

Such a good PR, I went to look for the demo link in the GH Checks and saw you put me a nice demo link in the description, and I was thinking wow this PR gets even better. And then I clicked on it and saw you'd used the deep linking stuff we did to take me straight to the page AND enable the scenario/cookies I needed to see it - super good!

File: ui/packages/consul-ui/app/router.js (hunk 3 above)
Comment (kaxcode) [reply ui/packages/consul-ui/app/router.js:203]: Renamed to be `nspace-rules`
